src/Rule/ForbidAssignmentNotMatchingVarDocRule.php

```diff
@@ -55,6 +60,10 @@ public function processNode(Node $node, Scope $scope): array
             $checkShapeOnly = true; // this is needed for example when phpstan-doctrine deduces nullable field, but you added WHERE IS NOT NULL
         }
 
+        if (mb_strpos($phpDoc->getText(), 'narrow') !== false) {
+            $allowNarrowing = true;
```

Comment:
The more I think about this the more I'm conviced we should keep only this option. The global one is imo not good idea (and using that in combination with this local change would be confusing). 

Would it be sufficient for you to keep only this "inplace behaviour configuration"? Meaning you need to use `/** @var Type narrow */` everywhere you know you are doing some narrowing.

Reply:
I like that idea, maybe I'd not encourage people to use the narrowing, so I'd not add the tip how to do it. I still see the narrowing as hack that should not be used.

```
Invalid var phpdoc of \${$variableName}. Cannot assign {$valueTypeString} to {$varPhpDocTypeString}

Invalid var phpdoc of \${$variableName}. Cannot narrow {$valueTypeString} to {$varPhpDocTypeString}.
```

:+1:

Reply:
PR updated